exercises/matching-brackets/example.jl

```diff
@@ -0,0 +1,33 @@
+function matching_brackets(a)
```

Comment:
General comments on this file:

- Please set your editor to replace tabs with 4 spaces to follow the style guide from the README.
- Perhaps use more descriptive variable names. It's hard to remember what `a`, `b` or `c` etc. mean when reading the code that's further down. For example, `b` and `c` could be `opening_brackets` and `closing_brackets`.
- Related to the previous point: `a` means a lot of different things throughout the function body. First it's the input string, then it's turned into an array and eventually it's filtered. This causes two issues:
  1. Readability: when reading the function, this confused me at L7-L9 because I still thought `a` was the input. 
  2. Performance: changing the type of variables should be avoided for performance reasons: https://docs.julialang.org/en/v1/manual/performance-tips/index.html#Avoid-changing-the-type-of-a-variable-1
  - I would suggest that you instead create multiple variables, or combine the steps into one and rename `a` to be more descriptive. The argument should probably be called `input` regardless

Reply:
Implemented the changes about coding style! 

On the 3rd comment, I don't think I've changed the type of `a` (now: `inputt`)? In lines 7-9, I check the length of a to see if it matches a special test cases which allows early exiting. My logic is that if the length of the brackets expression is not divisible by 2, then there is no chance of the expression being balanced.

Reply:
> On the 3rd comment, I don't think I've changed the type of a (now: inputt)?

It changed from String to an Array:

```julia
julia> function matching_brackets(a)
           @show a, typeof(a)

           a = split(a, "")
           @show a

           b,c = ["(", "[", "{"], [")", "]", "}"]

           a = filter(i -> (i in b || i in c), a)
           @show a
       end
matching_brackets (generic function with 1 method)

julia> matching_brackets("()")
(a, typeof(a)) = ("()", String)
a = SubString{String}["(", ")"]
a = SubString{String}["(", ")"]
2-element Array{SubString{String},1}:
 "("
 ")"
```

`@code_warntype` also shows it and marks the change red (`@_2` are line numbers):

![grafik](https://user-images.githubusercontent.com/20866761/71640721-c8db9280-2c8f-11ea-982b-986692835432.png)